From 700f0bb8ede0a894de4195ab62de94f89054a6a7 Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Mon, 23 Oct 2023 12:31:04 +0300
Subject: [PATCH] [SW-114485] zephyr: add new define for preboot die1

This patch adds a new define in zephyr make to select a specific board
configuration for preboot firmware to run in die1 arc1 which is used to
bringup D2D link (high speed) incase SPI_GW is not working.

Change-Id: Icb3b632b6e7b6f9f718de41536334e8e0fad15e1
---

diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index e4c7e54..d3e51ce 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -26,6 +26,7 @@
 BR2_PACKAGE_ZEPHYR_ARC_PID=y
 BR2_PACKAGE_ZEPHYR_ARC_EVENT=y
 BR2_PACKAGE_ZEPHYR_ARC_PREBOOT=y
+BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1=y
 BR2_PACKAGE_PREBOOT_ARC_EXT_ROM=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
 BR2_POST_IMAGE_FW_LOADER_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/agent-fw_loader.its"
diff --git a/package/zephyr/Config.in b/package/zephyr/Config.in
index 9db84af..813535e 100644
--- a/package/zephyr/Config.in
+++ b/package/zephyr/Config.in
@@ -50,3 +50,9 @@
 	select BR2_PACKAGE_ZEPHYR
 	help
 	  Preboot application on Zephyr ARC
+
+config BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1
+	bool "Zephyr ARC PREBOOT for Die1"
+	select BR2_PACKAGE_ZEPHYR
+	help
+	  Preboot application on Zephyr Die1 ARC
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 5915d88..5f49b9d 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -147,6 +147,22 @@
 $(foreach module,$(ZEPHYR_PREBOOT_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_PREBOOT_DIR),$(ZEPHYR_PREBOOT_NAME)))
 endif #BR2_PACKAGE_ZEPHYR_ARC_PREBOOT
 
+ifdef BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1
+ZEPHYR_COMPONENTS_LIST += preboot_d1
+ZEPHYR_PREBOOT_D1_DIR = build-preboot-d1
+ZEPHYR_PREBOOT_D1_NAME = zephyr_preboot_d1
+ZEPHYR_CONF_DEFAULT ?= $(ZEPHYR_PREBOOT_D1_DIR)
+ZEPHYR_CONF_OPTS_PREBOOT_D1 += -DBOARD=hl_$(DEVICE_NAME)_prebootd1_$(HL_TARGET)
+ZEPHYR_CONF_OPTS_PREBOOT_D1 += -DDEVICE=$(DEVICE_NAME)
+ZEPHYR_CONF_OPTS_PREBOOT_D1 += -DZEPHYR_BUILD_DIR=$(ZEPHYR_BUILDDIR)
+ZEPHYR_CONF_OPTS_PREBOOT_D1 += -DTARGET=$(HL_TARGET)
+ZEPHYR_PREBOOT_D1_SRCDIR =  $(EMBEDDED_DIR)/zephyr-apps/preboot/app/preboot_d1
+ZEPHYR_CONF_OPTS_PREBOOT_D1 += $(ZEPHYR_PREBOOT_D1_SRCDIR)
+ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(call CREATE_APP_VERSION,preboot,$(ZEPHYR_PREBOOT_D1_DIR)) $(ZEPHYR_CONF_OPTS_PREBOOT_D1) -B $(ZEPHYR_PREBOOT_D1_DIR)
+ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)/zephyr/zephyr.bin $(BINARIES_DIR)/preboot_d1.bin
+ZEPHYR_INSTALL_TARGET_CMDS += && $(INSTALL) -D -m 0644 $(@D)/$(ZEPHYR_PREBOOT_D1_DIR)/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_preboot_d1.elf
+endif #BR2_PACKAGE_ZEPHYR_ARC_PREBOOT_D1
 
 ifdef BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER
 ZEPHYR_COMPONENTS_LIST += fw_loader
